<div class="nav">

## Goals for this repository

* First, function as a place that helps me organize the various resources I used to learn about bioinformatic anaylysis of the microbiome and resistome.
* Eventually, create useful tutorials to help others learn about bioinformatics.

## Contribute
Have a tool or resource that you love and think would be helpful to others? Think you could help improve any of the text in this repository's documentation?
Just send a Pull Request with the details!

##### Disclaimer: The following guide is meant as a general overview of we, the Microbial Ecology Group, does to analyze metagenomic sequencing samples. We attempt to keep up with the latest in “best practices”, but these analytic tools continue to improve and the correct combination of tools and statistical methods is in perpetual debate. Please let us know if you have any questions, require clarification, or want to help us improve our methods!

## Overview
---

* Bioinformatic analyses
   * Level 1 - Core requirements for lab members
       * First step: Server log-on and navigation. 
         * Objectives: 
         * Resources:
           * Tutorial of basic commands (cd, ls, pwd, htop)
           * Videos from codeacademy or free educational account from data.camp? (could also work for R)
         * Deliverable: File that contains:
           * PATH to a kraken database
           * PATH to sequence reads
           * PATH to MEGARes
           * Find hidden file (based on hint with use of "ls")
       * Second step: Working with files
         * Objectives: 
           * Make directories/files and edit them.
           * Move, copy, remove files. 
           * Find size of directories.
           * Download files.
         * Resources:
           * Tutorial showing:
             * Make directory on "/scratch.global/"
             * Find and download a bacterial genome on local computer and use filezilla to upload to new directory on server.
             * Find the number of headers in the genome using grep (this code is provided)
             * Create file with your name and date on it.
             * Copy file and rename to something else. 
             * Dangers of RM *
         * Deliverable: Send file with:
             * PATH to created file
             * PATH to uploaded genome
   * Level 2 - Running AMR++
     * First step
         * Objective: Using screen to run commands. Modules on MSI.
         * Resources: 
   * Level 3 - Making custom nextflow pipelines/ running other tools
   * Level 4 - 
* Statistical analysis
  * Level 1 - 

