<div class="nav">

## Goals for this repository

* First, function as a place that helps me organize the various resources I used to learn about bioinformatic anaylysis of the microbiome and resistome.
* Eventually, create useful tutorials to help others learn about bioinformatics.
  - For now, the tutorials are roughdrafts written for specific projects that I aim to improve over time.

## Contribute
Have a tool or resource that you love and think would be helpful to others? Just send a Pull Request with the details!

## Table of Contents

- [Research groups](#research-teams)
- [Important papers](#important-papers)
- [Tools for your computer](#computer-tools)
- [Cheatsheets](#cheat-sheets)
- [Tutorials](#tutorials)
- [MEGARes](#megares)
- [AMR++](#amrplusplus)
  - [Pipeline overview](#pipeline-overview)
  - [QC trimming](#qc-trimming)
  - [Host filtering](#host-filtering)
  - [Resistome analysis](#resistome-analysis)
        - [Alignment to MEGARes](#alignment-megares)
        - [Filtering counts](#filtering-counts)
        - [Genes that require SNP confirmation](#snp-confirmation)
        - [Resistance Gene Identifier](#rgi)
        - [Analyzing SNPs](#analyzing-snp)
- [Statistical analysis](#stats)
  - [Overview](#stat-overview)
- [Whole-genome-sequencing](#WGS)


</div>

<main>
      
      
<article id="research-teams">
  
## Research groups


* [MEGlab](http://megares.meglab.org/) - "Our international multidisciplinary group of scientists and educators is addressing the issues of antimicrobial resistance (AMR) and microbial ecology in agriculture through research, outreach, and education. By characterizing risks related to AMR and microbial ecology, our center will identify agricultural production practices that are harmful and can be avoided, while also identifying and promoting production practices and interventions that are beneficial or do no harm to the ecosystem or public health. This will allow society to realize “sustainable intensification” of agriculture."
* [Noyes Lab](https://www.thenoyeslab.org/) - "We are a curious, driven and genuine group of individuals who unite around the common goal of scientific discovery. We cherish diversity -- of thoughts, backgrounds, experiences, skills and opinions.  We encourage (even relish!) respectful debate, and we value each others' quirks. We believe that livestock animals are an essential and valuable part of our local and global society.  We eschew the notion that livestock production is inherently at odds with human and environmental health.  And we use scientific discovery to advance livestock production in a "win-win" manner --  to benefit animals, humans and the planet. "

- Other groups with great websites

  
</article>



<article id="important-papers">
  
## Important Manuscripts

- Data analysis
  * [Consistent and correctable bias in metagenomic sequencing experiments](https://elifesciences.org/articles/46923) - Important biases in metagenomic sequencing projects
  
</article>






<article id="computer-tools">
  
## Tools for your computer

- Text editors
  * [Atom](https://atom.io/) - a lightweight, open-source editor featuring Teletype, a tool for real-time collaboration.   
  
</article>




      
<article id="cheat-sheets">
  
## Cheatsheets

* [Devhints](https://devhints.io/) - handy syntax & command reference for hundreds of languages and technologies

</article>




<article id="tutorials">
  
## Tutorials

- R packages
  * [Practical ggplot](https://wilkelab.org/practicalgg/) - Good tutorial for ussing ggplot to create a variety of figures

</article>






<article id="MEGARes">

## MEGARes

* [MEGARes](http://megares.meglab.org/) - The MEGARes database contains sequence data for approximately 8,000 hand-curated antimicrobial resistance genes accompanied by an annotation structure that is optimized for use with high throughput sequencing. The acyclical annotation graph of MEGARes allows for accurate, count-based, hierarchical statistical analysis of resistance at the population level, much like microbiome analysis, and is also designed to be used as a training database for the creation of statistical classifiers.

</article>






<article id="amrplusplus">
  
## AMR++

* [AMR++ v2.0](http://megares.meglab.org/amrplusplus/latest/html/v2/index.html) - The Microbial Ecology Group's bioinformatic pipeline for microbiome and resistome analysis.

</article>






<article id="stats">
  
## Statistical analysis

* [GustaME](https://sites.google.com/site/mb3gustame/) - Great resource that provides an overview to many common statistical methods for bioinformatics.

</article>






<article id="WGS">
  
## Whole-genome-sequencing

- Useful papers
  * [Comparing Genomic Variant Identification Methods](https://www.biorxiv.org/content/10.1101/733642v1) - Manuscript "Genomic variant identification methods alter Mycobacterium tuberculosis transmission inference"
  * [Population structure in time-scaled phylogenies](https://www.biorxiv.org/content/10.1101/704528v1) - Identification of hidden population structure in time-scaled phylogenies

</article>










