<div class="nav">

## Goals for this repository

* First, function as a place that helps me organize the various resources I used to learn about bioinformatic anaylysis of the microbiome and resistome.
* Eventually, create useful tutorials to help others learn about bioinformatics.
  - For now, the tutorials are roughdrafts written for specific projects that I aim to improve over time.

## Contribute
Have a tool or resource that you love and think would be helpful to others? Just send a Pull Request with the details!

## Table of Contents

- [Tools for your computer](#computer-tools)
- [Cheatsheets](#cheat-sheets)
- [MEGARes](#megares)
- [AMR++](#amrplusplus)
  - [Pipeline overview](#pipeline-overview)
  - [QC trimming](#qc-trimming)
  - [Host filtering](#host-filtering)
  - [Resistome analysis](#resistome-analysis)
        - [Alignment to MEGARes](#alignment-megares)
        - [Filtering counts](#filtering-counts)
        - [Genes that require SNP confirmation](#snp-confirmation)
        - [Resistance Gene Identifier](#rgi)
        - [Analyzing SNPs](#analyzing-snp)
- [Statistical analysis](#stats)
  - [Overview](#stat-overview)


</div>

<main>
      
      
      
<article id="computer-tools">
  
## Tools for your computer

- Text editors
  * [Atom](https://atom.io/) - a lightweight, open-source editor featuring Teletype, a tool for real-time collaboration.   
  
</article>




      
<article id="cheat-sheets">
  
## Cheatsheets

* [Devhints](https://devhints.io/) - handy syntax & command reference for hundreds of languages and technologies

</article>




<article id="MEGARes">

## MEGARes

* [MEGARes](http://megares.meglab.org/) - The MEGARes database contains sequence data for approximately 8,000 hand-curated antimicrobial resistance genes accompanied by an annotation structure that is optimized for use with high throughput sequencing. The acyclical annotation graph of MEGARes allows for accurate, count-based, hierarchical statistical analysis of resistance at the population level, much like microbiome analysis, and is also designed to be used as a training database for the creation of statistical classifiers.

</article>






<article id="amrplusplus">
  
## AMR++

* [AMR++ v2.0](http://megares.meglab.org/amrplusplus/latest/html/v2/index.html) - The Microbial Ecology Group's bioinformatic pipeline for microbiome and resistome analysis.

</article>






<article id="stats">
  
## Statistical analysis

* [GustaME](https://sites.google.com/site/mb3gustame/) - Great resource that provides an overview to many common statistical methods for bioinformatics.

</article>












